transport/src/main/java/io/netty/channel/DefaultChannelPromise.java

```diff
@@ -37,7 +39,7 @@
      *        the {@link Channel} associated with this future
      */
     public DefaultChannelPromise(Channel channel) {
-        this.channel = channel;
+        this.channel = checkNotNull(channel, "channel");
```

Comment:
Followup: this caused some tests in grpc to start failing I think.  We are probably at fault here, but just to let you know some code was doing this.

Reply:
@carl-mastrangelo thanks... that said I think it is a bug to pass in null :)